### PR TITLE
NCP: free the message from ot stack

### PR DESCRIPTION
**Pull request by luwannest**
The message from OT stack needs to be released after send out to the host.

### CODE REVIEW

File: src/ncp/ncp_base.cpp [hunk 1]
```diff
@@ -414,8 +416,8 @@ void NcpBase::HandleSendDone()
     if (mSendQueue.GetHead() != NULL)
     {
         Message &message(*mSendQueue.GetHead());
-        HandleDatagramFromStack(message);
         mSendQueue.Dequeue(message);
+        HandleDatagramFromStack(message);
```
Comment (darconeous) [inline src/ncp/ncp_base.cpp]: Doesn't dequeue release the message?

And, now that I'm looking at this, do we even want this line? It looks like it is trying to feed packets we sent out back into the stack.

Comment (luwannest) [inline src/ncp/ncp_base.cpp]: No. Dequeue  does not free the buffer.

I think that it handles the packets in the queue from the stack, and transmits to the host. The packets from the stack would enqueue, if the serial port is sending. Is my understanding right?

Comment (jwhui) [inline src/ncp/ncp_base.cpp]: Yes, Dequeue does not free the message, it just removes it from the queue.

It looks like `HandleDatagramFromStack` is responsible for taking a message buffer and sending it to the host using `spinel`.  That is the behavior we want, right?

Comment (darconeous) [inline src/ncp/ncp_base.cpp]: Yeah, but to we want to call it for datagrams which the stack itself emitted?

In any case, this particular discussion shouldn't block the rest of this commit, so I'll go ahead and approve this.

**Comment by darconeous** [comment]
👍

**Comment by jwhui** [comment]
LGTM 👍

**Pull request merged**